src/Microsoft.Security.Utilities.Core/HighConfidenceMicrosoftSecurityModels/IdentifiableKey.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Security.Utilities
+{
+    public abstract class IdentifiableKey : RegexPattern, IIdentifiableKey
+    {
+        private ISet<string>? _sniffLiterals;
+
+        public IdentifiableKey()
+        {
+            RotationPeriod = TimeSpan.FromDays(365 * 2);
+            DetectionMetadata = DetectionMetadata.Identifiable;
+        }
+
+        public abstract string Signature { get; }
+
+        public virtual uint KeyLength => 32;
+
+        public virtual bool EncodeForUrl => false;
+
+        public abstract IEnumerable<ulong> ChecksumSeeds { get; }
+
+        public override Tuple<string, string>? GetMatchIdAndName(string match)
+        {
+            foreach (ulong checksumSeed in ChecksumSeeds)
+            {
+                if (IdentifiableSecrets.ValidateBase64Key(match,
+                                                          checksumSeed,
+                                                          Signature))
+                {
+                    return new Tuple<string, string>(Id, Name);
+                }
+            }
+
+            return null;
+        }
+
+        public override ISet<string>? SniffLiterals
+        {
+            get
+            {
+                _sniffLiterals ??= new HashSet<string>(new[] { Signature });
+                return _sniffLiterals;
+            }
+
+            protected set => _sniffLiterals = value;
+        }
+
+        public override IEnumerable<string> GenerateTestExamples()
```

Comment:
>GenerateTestExamples [](http://example.com/codeflow?start=44&length=20)

Our test examples produce five examples per checksum seed which are obviously test things. This change prompted me to create this base class so I didn't create two copies of the same helper. This led me to address the technical debt of the other duplicated code in `Azure32ByteIdentifiableKey` and `Azure64ByteIdentifiableKey`.

`cccccccccccccccccccccccccccccccccTESTCi1lAI=1`